firewall.go

```diff
@@ -0,0 +1,76 @@
+package firewall
+
+import (
+	"github.com/libp2p/go-cidranger"
+	"net"
+	"net/http"
+	"time"
+)
+
+// BlockList inherits from cidranger.Ranger
+//credits github.com/libp2p/go-cidranger
+type BlockList struct {
+	cidranger.Ranger
+}
+
+func (bl *BlockList) appendIPBlocks(IPBlocks []string) error {
+	var (
+		err   error
+		IPNet *net.IPNet
+	)
+	for _, IPBlock := range IPBlocks {
+		if IPBlock != "" {
+			_, IPNet, err = net.ParseCIDR(IPBlock)
+			if err != nil {
+				return err
+			}
+			bl.Insert(cidranger.NewBasicRangerEntry(*IPNet))
+		}
+	}
+	return nil
+}
+
+// NewIPBlockList returns a new BlockList with inserted CIDR Ranges
+func NewIPBlockList(IPBlocks []string) (*BlockList, error) {
+	bl := BlockList{cidranger.NewPCTrieRanger()}
```

Comment:
change this to &BlockList{}
the append will not make a new object below on append

Reply:
ohhh, got it, done :+1: